sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1128,15 +1255,37 @@ public static class Write extends Mutate<Entity> {
         @Nullable String localhost,
         boolean throttleRampup,
         ValueProvider<Integer> hintNumWorkers) {
-      super(projectId, localhost, new UpsertFn(), throttleRampup, hintNumWorkers);
+      super(projectId, DEFAULT_DATABASE, localhost, new UpsertFn(), throttleRampup, hintNumWorkers);
     }
 
-    /** Returns a new {@link Write} that writes to the Cloud Datastore for the specified project. */
+    /**
+     * Note that {@code projectId} is only {@code @Nullable} as a matter of build order, but if it
+     * is {@code null} at instantiation time, an error will be thrown.
+     */
+    Write(
+        @Nullable ValueProvider<String> projectId,
+        String databaseId,
+        @Nullable String localhost,
+        boolean throttleRampup,
+        ValueProvider<Integer> hintNumWorkers) {
+      super(projectId, databaseId, localhost, new UpsertFn(), throttleRampup, hintNumWorkers);
+    }
+
+    /** Returns a new {@link Write} that writes to the Cloud Datastore for the default database. */
     public Write withProjectId(String projectId) {
       checkArgument(projectId != null, "projectId can not be null");
       return withProjectId(StaticValueProvider.of(projectId));
     }
 
+    /**
+     * Returns a new {@link Write} that writes to the Cloud Datastore for the specified database.
+     */
+    public Write withProjectAndDatabaseId(String projectId, String databaseId) {
```

Comment:
generally we don't have withX methods that pass two parameters. it is probably better to set up another method that just passes in the databaseId.